File: Project_Euler/3/three.cpp
#include <iostream>
using namespace std;

int check_prime(long num);
int get_factors(long num);

int main() {
	long var;
	cin >> var;
	get_factors(var);
}

int check_prime(long num)
{
	int flag = 0;
		// System.out.println(num/2);
	for(int i=2;i<=num/2;i++){
		if(num%i==0)
		{
			flag = 1;
			break;
		}
	}
	if (flag == 0){
		return flag;
			// no is prime
	}
	else{
		return flag;
			// no is not prime
	}
}

int get_factors(long num)
{
	int flag = 0;
	int i = 2;
	while(i<=num/2){
		if(num%i == 0 && check_prime(i) == 0){ //checked if the number is both 
			cout<<i<<"this";
		}
		i++;
	}
	return 0;
}

File: Project_Euler/3/three.py
from functools import reduce
def check_prime(num):
	if (num == 2):
		flag = 0
	flag = 0
	for i in range(2,int(num/2)):
		if (num%i==0):
			flag = 1
			break
	return flag

def get_factors(num):
	a = []
	for i in range(2,int(num/2)):
		if (num%i ==0 and check_prime(i) == 0):
			a.append(i)
			new_a = reduce(lambda x, y: x*y,a)
			if new_a == num:
				break
			else:
				continue
	print('\nThe largest prime factor is',max(a))
	return 0

var = int(input("Insert number "))
get_factors(var)
